src/Ilios/CoreBundle/Controller/AuthenticationController.php

```diff
@@ -47,31 +49,62 @@ public function postAction(Request $request)
     {
         try {
             $handler = $this->container->get('ilioscore.authentication.handler');
-            $data = $this->getPostData($request);
+            $arr = $this->getPostData($request);
+            $count = count($arr);
+            if ($count > 500) {
+                throw new BadRequestHttpException("Maximum of 500 users can be created.  You sent " . $count);
+            }
 
-            if (!empty($data['password']) && !empty($data['user'])) {
-                $userManager = $this->container->get('ilioscore.user.manager');
-                $user = $userManager->findOneBy(['id' => $data['user']]);
-                if ($user) {
-                    $encoder = $this->container->get('security.password_encoder');
-                    $encodedPassword = $encoder->encodePassword($user, $data['password']);
-                    $data['passwordBcrypt'] = $encodedPassword;
-                }
+            $unsavedItems = [];
+            $manager = $this->container->get('ilioscore.authentication.manager');
+            $encoder = $this->container->get('security.password_encoder');
+            $userManager = $this->container->get('ilioscore.user.manager');
+
+            $needingHashedPassword = array_filter($arr, function ($arr) {
+                return (!empty($arr['password']) && !empty($arr['user']));
+            });
+            $userIdsForHashing = array_map(function ($arr) {
+                return $arr['user'];
+            }, $needingHashedPassword);
+            //prefetch all the users we need for hashing
+            $users = [];
+            foreach ($userManager->findBy(['id' => $userIdsForHashing]) as $user) {
+                $users[$user->getId()] = $user;
             }
-            //unset the password here in case it is NULL and didn't satisy the above condition
-            unset($data['password']);
 
-            $authentication = $handler->post($data);
+            foreach ($arr as $data) {
+                if (!empty($data['password']) && !empty($data['user'])) {
+                    $user = $users[$data['user']];
+                    if ($user) {
+                        $encodedPassword = $encoder->encodePassword($user, $data['password']);
+                        $data['passwordBcrypt'] = $encodedPassword;
```

Comment:
this won't work. you can't directly manipulate array items in a `foreach` loop like that, either pass `$data` by reference or change this to a `for` loop construct.